pkg/api/dataproxy.go

```diff
@@ -54,6 +54,15 @@ func (hs *HTTPServer) ProxyDataSourceRequest(c *m.ReqContext) {
 	}
 
 	proxyPath := c.Params("*")
+
+	// Check for a trailing slash
+	if len(proxyPath) > 1 {
+		path := c.Req.URL.Path
+		if path[len(path)-1] == '/' && path[len(path)-2] != '/' {
```

Comment:
why are you checking if eiher last or next to last is forward slash, should it not check if last is forward slash in original path, and if it's not forward slash in proxyPath  (that way if it's fixed in macaron it wont introduce a bug here)

Reply:
good catch -- i changed it